yaptide/batch/watcher.py

```diff
@@ -0,0 +1,109 @@
+from pathlib import Path
+import re
+import time
+import argparse
+import signal
+
+
+RUN_MATCH = r"\bPrimary particle no.\s*\d*\s*ETR:\s*\d*\s*hour.*\d*\s*minute.*\d*\s*second.*\b"
+COMPLETE_MATCH = r"\bRun time:\s*\d*\s*hour.*\d*\s*minute.*\d*\s*second.*\b"
+REQUESTED_MATCH = r"\bRequested number of primaries NSTAT"
+TIMEOUT_MATCH = r"\bTimeout occured"
+
+
+def log_generator(thefile, timeout: int = 3600):
+    """
+    Generator equivalent to `tail -f` Linux command.
+    Yields new lines appended to the end of the file.
+    Main purpose is monitoring of the log files
+    """
+    sleep_counter = 0
+    while True:
+        line = thefile.readline()
+        if not line:
+            time.sleep(1)
+            sleep_counter += 1
+            if sleep_counter >= timeout:
+                yield "Timeout occured"
```

Comment:
```suggestion
                return "Timeout occured"
```

shouldn't we ensure that the generator is exhausted ?